pkg/query-service/app/clickhouseReader/reader.go

```diff
@@ -1497,7 +1497,11 @@ func buildQueryWithTagParams(ctx context.Context, tags []model.TagQuery) (string
 		tagMapType := ""
 		switch item.(type) {
 		case model.TagQueryString:
-			tagMapType = constants.StringTagMapCol
+			if item.GetTagType() == model.ResourceAttributeTagType {
+				tagMapType = constants.ResourceTagMapCol
+			} else {
+				tagMapType = constants.StringTagMapCol
+			}
 		case model.TagQueryNumber:
 			tagMapType = constants.NumberTagMapCol
 		case model.TagQueryBool:
```

Comment:
I would refactor this to get the tagMap type from the `item` itself since the only thing written inside the case is `tagMapType` variable assignment. That would be cleaner than `switch` and `if`.